soc/rapidsilicon/virgo/soc.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Rapid Silicon.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * VIRGO SoC specific interrupt map and mask functions
+ */
+
+#include <zephyr/kernel.h>
+
+#define SOC_IRQ_MASK_OFFSET 0
+#define SOC_IRQ_MAP_OFFSET  16
+
+#define SOC_IRQ_DISABLE     0x0
+#define SOC_IRQ_ENABLE      0x1
+
+#define SOC_IRQ_MAP_TO_NONE 0x0
+#define SOC_IRQ_MAP_TO_BCPU 0x1
+#define SOC_IRQ_MAP_TO_RSVD 0x2
+#define SOC_IRQ_MAP_TO_FPGA 0x4
+#define SOC_IRQ_MAP_MASK    0x7
+
+enum map_mask_control_irq_id {
+  IRQ_ID_BCPU_WDT = -1,
+  IRQ_ID_GPT = 0,
+  IRQ_ID_RESERVED1 = 1,
+  IRQ_ID_RESERVED2 = 2,
+  IRQ_ID_UART0 = 3,
+  IRQ_ID_RESERVED4 = 4,
+  IRQ_ID_SPI = 5,
+  IRQ_ID_I2C = 6,
+  IRQ_ID_GPIO = 7,
+  IRQ_ID_SYSTEM_DMA = 8,
+  IRQ_ID_RESERVED9 = 9,
+  // Mailboxes
+  IRQ_ID_RESERVED10 = 10,
+  IRQ_ID_RESERVED11 = 11,
+  IRQ_ID_RESERVED12 = 12,
+  IRQ_ID_RESERVED13 = 13,
+  // Encryption engine
+  IRQ_ID_PUFCC = 14,
+  // FPGA GENERATED
+  IRQ_ID_FPGAGEN0 = 15,   // Not mapped to FPGA
```

Comment:
```suggestion
  IRQ_ID_FPGAGEN0 = 15,   // Not mapped to FPGA
```
"// Not mapped to FPGA" maybe true for prototype, but for the actual chip the mapping should be there and the FPGA should be able to interrupt the BCPU. Since this is related to SOC and not platform (proto or actual board), we should probably remove this comment

Reply:
Done